src/test/java/com/codeus/winter/config/DefaultBeanFactoryTest.java

```diff
@@ -628,6 +628,22 @@ void testFailInitializeBeanWithPrototypeScope() {
 
     }
 
+    @Test
+    @DisplayName("Should initialize a bean with multiple candidates using @Qualifier annotation")
+    void testShouldInitializeBeanWithMultipleCandidatesUsingQualifier() {
+        BeanDefinition beanDefinitionWithQualifierAnnotation = new BeanDefinitionImpl();
+        beanDefinitionWithQualifierAnnotation.setBeanClassName("com.codeus.winter.test.BeanWithQualifierAnnotation");
+        HashMap<String, BeanDefinition> beanDefinitionHashMap = new HashMap<>();
+        beanDefinitionHashMap.put("BeanA", beanDefinitionA);
+        beanDefinitionHashMap.put("BeanE", beanDefinitionE);
+        beanDefinitionHashMap.put("BeanWithQualifier", beanDefinitionWithQualifierAnnotation);
+
+        DefaultBeanFactory factory = new DefaultBeanFactory(beanDefinitionHashMap);
+        factory.initializeBeans();
+
+        assertNotNull(factory.getBean(BeanWithQualifierAnnotation.class));
```

Comment:
Let's also assert that `BeanWithQualifierAnnotation`'s `Common common` is not null and has type `BeanA`.

Reply:
Added